packages/shorebird_cli/lib/src/network_checker.dart

```diff
@@ -36,4 +42,35 @@ class NetworkChecker {
       }
     }
   }
+
+  /// Uploads a file to GCP to measure upload speed.
+  Future<void> performGCPSpeedTest() async {
+    // Test with a 5MB file.
+    const uploadMBs = 5;
+    const fileSize = uploadMBs * 1000 * 1000;
+    final tempDir = Directory.systemTemp.createTempSync();
+    final testFile = File(p.join(tempDir.path, 'speed_test_file'))
+      ..writeAsBytesSync(ByteData(fileSize).buffer.asUint8List());
+
+    final progress = logger.progress('Performing GCP speed test');
+
+    final uri = await codePushClientWrapper.getGCPSpeedTestUrl();
+    final start = DateTime.now();
+    final file = await http.MultipartFile.fromPath('file', testFile.path);
+    final uploadRequest = http.MultipartRequest('POST', uri)..files.add(file);
+    final uploadResponse = await httpClient.send(uploadRequest);
```

Comment:
should we set a timeout so that we're not hanging out waiting forever in cases where a user can't upload?

Reply:
Done